apps/graphql/src/apps/itinerary/dataloaders/Itineraries.js

```diff
@@ -71,20 +77,59 @@ const addReturnSearchQueryParams = inboundDate => {
   };
 };
 
+const parseSlugsToIds = async (
+  location: SearchLocation,
+): Promise<SearchLocation> => {
+  if (!location.slugs) return location;
+
+  const results: $ReadOnlyArray<ApiResponseLocation> = await Promise.all(
+    location.slugs.map(slug => fetch(`/locations/slug?term=${slug}`)),
+  );
+
+  const getFirstValidID = (locations: $ReadOnlyArray<ApiLocation>): ?string =>
+    locations.filter(location => location.id)[0].id;
+
+  const IDs: string[] = results.map(
+    result => getFirstValidID(result.locations) ?? '',
+  );
+
+  return { ...location, ids: IDs.filter(id => id) };
+};
+
 const fetchItineraries = async (
   parameters: $ReadOnlyArray<ItinerariesReturnSearchParameters>,
 ) => {
-  const results: $ReadOnlyArray<ApiResponseType> = await Promise.all(
-    parameters.map(params => {
-      return fetch(`/v2/search?${qs.stringify(parseParameters(params))}`);
+  const results: $ReadOnlyArray<ApiResponseSearch> = await Promise.all(
+    parameters.map(async params => {
+      let { origin, destination } = params.itinerary;
+
+      origin = slugsNeedParsing(origin)
+        ? await parseSlugsToIds(origin)
+        : unmaskReplaceIds(origin);
+
+      destination =
+        destination && slugsNeedParsing(destination)
+          ? await parseSlugsToIds(destination)
+          : unmaskReplaceIds(destination);
```

Comment:
Ok, it is better, the only thing I don't like is that you still overwrite the original object.

Please do `const parsedOrigin` and `const parsedDestination` instead. Then the PR should be ready to be merged.

Reply:
Changed! Yes it's a little bit ugly, it would be much nicer if the `Tequila API` returned also the opaque IDs so we could move it back to `parseParameters` and not have to do this ID tweaking...

Reply:
Well, I still don't understand the use case for this, why do we need this query?

Reply:
@tbergq We just talked about it with Albert and this part is not longer relevant. After ve merged everything to `RouteStop` in #606 we don't need this extra parsing and everything (slug included) is solved by Location query now.